CorsixTH/Lua/dialogs/resizables/adviser_history.lua

```diff
@@ -0,0 +1,181 @@
+--[[ Copyright (c) 2025 Damian "ShiroAka"
+
+Permission is hereby granted, free of charge, to any person obtaining a copy of
+this software and associated documentation files (the "Software"), to deal in
+the Software without restriction, including without limitation the rights to
+use, copy, modify, merge, publish, distribute, sublicense, and/or sell copies
+of the Software, and to permit persons to whom the Software is furnished to do
+so, subject to the following conditions:
+
+The above copyright notice and this permission notice shall be included in all
+copies or substantial portions of the Software.
+
+THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND, EXPRESS OR
+IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF MERCHANTABILITY,
+FITNESS FOR A PARTICULAR PURPOSE AND NONINFRINGEMENT. IN NO EVENT SHALL THE
+AUTHORS OR COPYRIGHT HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER
+LIABILITY, WHETHER IN AN ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING FROM,
+OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN THE
+SOFTWARE. --]]
+
+--! Adviser message history
+class "UIAdviserHistory" (UIResizable)
+
+---@type UIAdviserHistory
+local UIAdviserHistory = _G["UIAdviserHistory"]
+
+local col_bg = { red = 60, green = 174, blue = 201, }
+local col_highlight = { red = 47, green = 154, blue = 190, }
+local col_shadow = { red = 36, green = 138, blue = 158 }
+local col_delete_button = { red = 219, green = 36, blue = 36 }
+
+-- Timer, used to update the dialog only once per X ticks to reduce the burden.
+local ticks_to_skip = 5
+local tick_timer = ticks_to_skip
+
+local row_height = 40
```

Comment:
It might be possible to make your row heights dynamic with some help from the draw functions on text (I think UIInformation and UIConfirmation do this). But I need to look into this patch deeper as well.